packages/nvidia_nat_mcp/src/nat/plugins/mcp/client_impl.py

```
@@ -75,103 +98,219 @@ def mcp_client_transport(self, transport: str | None):
+                    inactive_sessions.append(session_id)
+
+            for session_id in inactive_sessions:
+                try:
+                    logger.info("Cleaning up inactive session client: %s", truncate_session_id(session_id))
+                    session_data = self._sessions[session_id]
+                    # Close the client connection
+                    await session_data.client.__aexit__(None, None, None)
+                    del self._sessions[session_id]
+                    logger.info("Cleaned up inactive session client: %s", truncate_session_id(session_id))
+                    logger.info(" Total sessions: %d", len(self._sessions))
+                except Exception as e:
```

Ruff (0.13.2)

Do not catch blind exception: `Exception`

(BLE001)